apollo-federation/src/query_plan/conditions.rs

```diff
@@ -70,42 +71,52 @@ impl Conditions {
         }
     }
 
+    /// Parse @skip and @include conditions from a directive list.
     pub(crate) fn from_directives(directives: &DirectiveList) -> Result<Self, FederationError> {
         let mut variables = IndexMap::default();
-        for directive in directives.iter() {
-            let negated = match directive.name.as_str() {
-                "include" => false,
-                "skip" => true,
-                _ => continue,
+
+        if let Some(skip) = directives.get("skip") {
```

Comment:
It appears that the `directives` may have multiple skip/include directive applications. This will handle only one of each type.